group_vars/all.yml

```diff
@@ -61,18 +61,28 @@ unicorn_timeout: 60
 #----------------------------------------------------------------------
 # nginx config
 #
-# Use of `deferred` delays handling a request until the client sent data.
+# Some `listen` params can only be applied once per port, so are applied in the default_servers.
+# Use of `deferred` param delays handling a request until the client sent data.
 
 nginx_sites_available:
-  default:
+  default_https:
     - |
-      listen 80 default_server deferred;
-      listen [::]:80 default_server deferred;
-      server_name _;
+      # Default port settings
+      listen 80 deferred default_server;
+      listen [::]:80 deferred;
+      listen 443 ssl deferred default_server;
+      listen [::]:443 ssl deferred default_server;
+
+      # Defaults for any server_names not specified elsewhere
+      # nginx requires these parameters for ssl ports, even though we're not responding:
+      ssl_certificate      /etc/letsencrypt/live/{{ domain }}/fullchain.pem;
+      ssl_certificate_key  /etc/letsencrypt/live/{{ domain }}/privkey.pem;
```

Comment:
If this is responding to an unconfigured domain, shall we use the system's self-signed cert?
```
include snippets/snakeoil.conf;
```

Reply:
Thanks, I didn't know there was a default cert!

Reply:
Done, great suggestion thanks.